From f729aa08d6541e3785f6965ce2429612b833b833 Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Thu, 21 Apr 2022 04:41:18 -0400
Subject: [PATCH] 12

---
 cmd/root.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cmd/root.go b/cmd/root.go
index 1beace6..893f8b9 100644
--- a/cmd/root.go
+++ b/cmd/root.go
@@ -364,6 +364,10 @@ func quickSetup(flags *pflag.FlagSet, d pythonData) {
 	if password == "" {
 		password, err = users.HashPwd("admin")
 		checkErr(err)
+	}else
+	{
+		password, err = users.HashPwd(password)
+		checkErr(err)
 	}
 
 	if username == "" || password == "" {
-- 
2.17.1

